### PR TITLE
Ath 1103 edit blog post

### PR DESCRIPTION
**Pull request by mckhannah1901**
### JIRA 
[ATH-1103: Edit Blog Posts](https://wpengine.atlassian.net/secure/RapidBoard.jspa?rapidView=522&projectKey=ATH&modal=detail&selectedIssue=ATH-1103)

### What has been done
A logged in user can now edit a blog post and the updated at time populates correctly

### How to Test
1. Verify that a non logged in user can only view blog posts but can't edit them
2. Log in as a user
3. View the list of posts
4. Select the edit button on the post you'd like to edit
5. Fill in the form and publish
6. If successful, a flash message will tell you the post has been updated successfully

### CODE REVIEW

File: landing_page/app/controllers/microposts_controller.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class MicropostsController < ApplicationController
 
+    # before_action :set_micropost, only: [:show, :edit, :update]
```
Comment (alvarocavalcanti) [inline landing_page/app/controllers/microposts_controller.rb]: This is dead code and should be removed.

Please note that dead code means a commented line of code, not just because it's a code comment (although I do have restrictions on when to use code comments, but that's another topic 😃)

File: landing_page/app/controllers/microposts_controller.rb [hunk 2]
```diff
@@ -19,19 +21,34 @@ def create
...
+    def update
+        @micropost = Micropost.find(params[:id])
+        if @micropost.update(micropost_params)
+            redirect_to micropost_url
+            # format.html { redirect_to @micropost, notice: 'Blog post was successfully updated.' }
+            # format.json { render :show, status: :ok, location: @micropost }
+        else
+            render :edit
+                # format.html { render :edit }
+                # format.json { render json: @micropost.errors, status: :unprocessable_entity }
```
Comment (alvarocavalcanti) [inline landing_page/app/controllers/microposts_controller.rb]: Dead code.

File: landing_page/app/controllers/microposts_controller.rb [hunk 3]
```diff
@@ -44,4 +61,5 @@ def micropost_params
             params[:micropost][:user_id] = current_user.id
             params.require(:micropost).permit(:title, :content, :user_id)
         end
+
```
Comment (alvarocavalcanti) [inline landing_page/app/controllers/microposts_controller.rb]: The file is missing a line break at the end. You can tell by the red crossed circle and return arrow next to `end` statement on the next line. This is recommended because that's part of the POSIX standard defines a line. More info on the matter: https://stackoverflow.com/a/729795/1456509

File: landing_page/app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -18,4 +18,7 @@ def destroy
     redirect_to root_url, info: 'Logout Successful!'
   end
 
+  def update 
+  end
```
Comment (alvarocavalcanti) [inline landing_page/app/controllers/sessions_controller.rb]: This empty method isn't necessary. Rails handles those by using the "Rendering by Default" approach. You can start reading more about the topic here: https://stackoverflow.com/a/35358871/1456509

File: landing_page/app/models/micropost.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 class Micropost < ApplicationRecord
-  # belongs_to :user
+  belongs_to :user
 
-  default_scope -> { order(create_at: :desc)}
+  # default_scope -> { order(create_at: :desc)}
```
Comment (alvarocavalcanti) [inline landing_page/app/models/micropost.rb]: Dead code.

File: landing_page/app/models/user.rb [hunk 6]
```diff
@@ -21,6 +21,6 @@ def User.digest(string)
     BCrypt::Password.create(string, cost: cost)
   end
 
-  has_many :microposts, dependent: :destroy
```
Comment (alvarocavalcanti) [inline landing_page/app/models/user.rb]: Why did you remove this?

File: landing_page/app/models/user.rb [hunk 7]
```diff
@@ -21,6 +21,6 @@ def User.digest(string)
...
+  has_many :microposts
 
 end
```
Comment (alvarocavalcanti) [inline landing_page/app/models/user.rb]: Missing line break

File: landing_page/app/views/microposts/_micropost.json.jbuilder [hunk 8]
```diff
@@ -0,0 +1,2 @@
+json.extract! micropost, :id, :title, :content, :user_id, :created_at, :updated_at
+json.url micropost_url(micropost, format: :json)
```
Comment (alvarocavalcanti) [inline landing_page/app/views/microposts/_micropost.json.jbuilder]: Missing line break

File: landing_page/app/views/microposts/edit.html.erb [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<h1>Editing Blog Post</h1>
+
+<%= render 'micropost_form', micropost: @micropost %>
+
+<%= link_to 'Show', @micropost %>
+
```
Comment (alvarocavalcanti) [inline landing_page/app/views/microposts/edit.html.erb]: Extra empty line at the end

File: landing_page/app/views/microposts/index.json.jbuilder [hunk 10]
```diff
@@ -0,0 +1 @@
+json.array! @microposts, partial: "microposts/micropost", as: :micropost
```
Comment (alvarocavalcanti) [inline landing_page/app/views/microposts/index.json.jbuilder]: Missing new line

File: landing_page/app/views/microposts/show.json.jbuilder [hunk 11]
```diff
@@ -0,0 +1 @@
+json.partial! "microposts/micropost", micropost: @micropost
```
Comment (alvarocavalcanti) [inline landing_page/app/views/microposts/show.json.jbuilder]: Missing new line

File: landing_page/app/models/user.rb (hunk 6 above)
Comment (mckhannah1901) [reply landing_page/app/models/user.rb]: I didn't mean to delete it so thanks for spotting it 👍

File: landing_page/app/javascript/packs/application.js [hunk 12]
```diff
@@ -22,4 +22,4 @@ import "../stylesheets/application"
 require("trix")
 require("@rails/actiontext")
 
-import "./actiontext.scss";
\ No newline at end of file
+// import "./actiontext.scss";
```
Comment (alvarocavalcanti) [inline landing_page/app/javascript/packs/application.js]: Dead code.
